app/controllers/v0/upload_supporting_evidences_controller.rb

```diff
@@ -4,23 +4,69 @@
 
 module V0
   class UploadSupportingEvidencesController < ApplicationController
-    include FormAttachmentCreate
     extend Logging::ThirdPartyTransaction::MethodWrapper
     service_tag 'disability-application'
 
-    FORM_ATTACHMENT_MODEL = SupportingEvidenceAttachment
+    VALID_FILE_CLASSES = [ActionDispatch::Http::UploadedFile, Rack::Test::UploadedFile].freeze
 
     wrap_with_logging(
       :save_attachment_to_cloud!,
       additional_class_logs: {
         form: '526ez supporting evidence attachment',
-        action: "upload: #{FORM_ATTACHMENT_MODEL}",
+        action: 'Uploaded user-provided attachment to S3',
         upstream: 'User provided file',
         downstream: "S3 bucket: #{Settings.evss.s3.bucket}"
       },
       additional_instance_logs: {
         user_uuid: %i[current_user account_uuid]
       }
     )
+
+    def create
+      validate_file_upload_class!
+      save_attachment_to_cloud!(form_attachment)
+      save_attachment_to_db!
+      save_lighthouse_attachment
+
+      render(json: form_attachment)
+    end
+
+    private
+
+    def validate_file_upload_class!
+      file_class = attachment_params[:file_data].class
+
+      unless VALID_FILE_CLASSES.include?(file_class)
+        raise Common::Exceptions::InvalidFieldValue.new('file_data', file_class)
+      end
+    end
+
+    def save_attachment_to_cloud!(attachment)
+      attachment.set_file_data!(attachment_params[:file_data], attachment_params[:password])
+    end
+
+    def save_attachment_to_db!
+      form_attachment.save!
+    end
+
+    def form_attachment
+      @form_attachment ||= SupportingEvidenceAttachment.new
+    end
+
+    # Form526 Document Uploads are being migrated to post to Lighthouse instead of EVSS
+    # LighthouseSupportingEvidenceAttachment will replace SupportingEvidenceAttachment as the model for these documents
+    # To ensure attachments uploaded before Lighthouse support is fully enabled are retreiveable on Form 526 Submissions
+    # that occur _after_ the feature is enabled, we need to create a LighthouseSupportingEvidenceAttachment record in
+    # addition to a SupportingEvidenceAttachment for now
+    def save_lighthouse_attachment
+      lighthouse_attachment = LighthouseSupportingEvidenceAttachment.new
```

Comment:
I might be missing something, but would it be possible to plan out a data migration once this gets merged instead of double-creating attachments now? It seems like file naming convention is identical, so I'm not clear on why this is needed now.